src/osd/scrubber/scrub_machine.cc

```diff
@@ -917,202 +958,32 @@ ReplicaIdle::ReplicaIdle(my_context ctx)
-  return discard_event();
-}
 
-
-// ---------------- ReplicaIdle/ReplicaWaitingReservation ---------------------------
-
-ReplicaWaitingReservation::ReplicaWaitingReservation(my_context ctx)
-    : my_base(ctx)
-    , NamedSimply(
-	  context<ScrubMachine>().m_scrbr,
-	  "ReplicaActive/ReplicaIdle/ReplicaWaitingReservation")
-{
-  dout(10)
-      << "-- state -->> ReplicaActive/ReplicaIdle/ReplicaWaitingReservation"
-      << dendl;
-}
-
-sc::result ReplicaWaitingReservation::react(const ReserverGranted& ev)
-{
-  DECLARE_LOCALS;  // 'scrbr' & 'pg_id' aliases
-  dout(10) << fmt::format(
-		  "ReplicaWaitingReservation::react(const ReserverGranted&): "
-		  "event:{}",
-		  ev)
-	   << dendl;
-  if (context<ReplicaActive>().granted_by_reserver(ev.value)) {
-    return transit<ReplicaReserved>();
-  }
-  return discard_event();
-}
-
-sc::result ReplicaWaitingReservation::react(const ReplicaRelease& ev)
-{
-  DECLARE_LOCALS;  // 'scrbr' & 'pg_id' aliases
-  dout(10) << "ReplicaWaitingReservation::react(const ReplicaRelease&)"
-	   << dendl;
-  // must cancel the queued reservation request
-  context<ReplicaActive>().on_release(ev);
-  return transit<ReplicaUnreserved>();
-}
-
-sc::result ReplicaWaitingReservation::react(const StartReplica& ev)
-{
-  DECLARE_LOCALS;  // 'scrbr' & 'pg_id' aliases
-  dout(10) << "ReplicaWaitingReservation::react(const StartReplica&)" << dendl;
-
-  // this shouldn't happen. We will handle it, but will also log an error.
-  scrbr->get_clog()->error() << fmt::format(
-      "osd.{} pg[{}]: new chunk request while still waiting for "
-      "reservation",
-      scrbr->get_whoami(), scrbr->get_spgid());
-  context<ReplicaActive>().clear_remote_reservation(true);
-  clear_shallow_history<ReplicaIdle, 0>();
-  post_event(ReplicaPushesUpd{});
-  return transit<ReplicaActiveOp>();
-}
-
-sc::result ReplicaWaitingReservation::react(const ReplicaReserveReq& ev)
-{
-  DECLARE_LOCALS;  // 'scrbr' & 'pg_id' aliases
-  dout(10) << "ReplicaWaitingReservation::react(const ReplicaReserveReq&)"
-	   << dendl;
-  // this shouldn't happen. We will handle it, but will also log an error.
-  scrbr->get_clog()->error() << fmt::format(
-      "osd.{} pg[{}]: reservation requested while previous is pending",
-      scrbr->get_whoami(), scrbr->get_spgid());
-  // cancel the existing reservation, and re-request
-  context<ReplicaActive>().clear_remote_reservation(true);
-  post_event(ev);
-  return transit<ReplicaUnreserved>();
-}
-
-
-// ---------------- ReplicaIdle/ReplicaReserved ---------------------------
-
-ReplicaReserved::ReplicaReserved(my_context ctx)
-    : my_base(ctx)
-    , NamedSimply(
-	  context<ScrubMachine>().m_scrbr,
-	  "ReplicaActive/ReplicaIdle/ReplicaReserved")
+void ReplicaIdle::reset_ignored(const FullReset&)
 {
-  dout(10) << "-- state -->> ReplicaActive/ReplicaIdle/ReplicaReserved"
+  dout(10) << "ReplicaIdle::react(const FullReset&): FullReset ignored"
 	   << dendl;
```

Comment:
Shouldn't this just reenter ReplicaActive as with every other reset?  If we're holding/waiting for a reservation, shouldn't it be cleared?

Reply:
Now that you mention it... FullReset is never expected on the Replica.
I will combine the reactions of Idle vs Op into one - of logging and ignoring.
The 'abort & reenter ReplicaActive' reaction should never be required here.

Reply:
i.e. - I will be changing it now to what it was before this PR...